### ISSUE TITLE
Kernel version dependency

### ISSUE THREAD

**Issue by blackPantherOS** [darlinghq/darling#639]
Hello, The darling is the true redemption of breaking down obstacles to Linux usage when the gui will be fully implemented as well, Thank You!

The first darling RPM package (available here: ftp://ftp.blackpantheros.eu/blackPanther/OS/All/v18.x/64bit/devel/darling-0.20200119-1bP.x86_64.rpm) created successfully for blackPanther OS, but lkm build failed.
Which the last compatible kernel version requires for lkm module build?
Thanks 

`
make -C /lib/modules/5.1.15-desktop-1bP/build M=/usr/src/RPM/BUILD/darling/src/lkm modules
make[5]: Entering directory '/usr/src/kernel-5.1.15-desktop-1bP'
Running kernel version is 5.1.15-desktop-1bP
Invoked by kernel build system, building for 5.1.15-desktop-1bP
  CC [M]  /usr/src/RPM/BUILD/darling/src/lkm/osfmk/ipc/ipc_entry.o
  CC [M]  /usr/src/RPM/BUILD/darling/src/lkm/osfmk/ipc/ipc_hash.o
In file included from /usr/src/RPM/BUILD/darling/src/lkm/bsd/kern/ast.h:37,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/kern/processor.h:75,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/i386/cpu_data.h:41,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/machine/cpu_data.h:32,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/kern/cpu_data.h:42,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/kern/task.h:110,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/ipc/ipc_space.h:88,
                 from /usr/src/RPM/BUILD/darling/src/lkm/osfmk/ipc/ipc_hash.c:76:
/usr/src/RPM/BUILD/darling/src/lkm/osfmk/kern/thread.h:434:2: error: unknown type name ‘task_watch_t’
  task_watch_t * taskwatch;  /* task watch */
  ^~~~~~~~~~~~
`

**Edit: 
Ok, i unset CONFIG_EMBEDDED and kernel module build successful as well..**

**Comment by ahyattdev** [comment]
This issue has been fixed, but we haven’t made RPMs yet. Will publish updated ones today

**Comment by blackPantherOS** [comment]
Fine thank you, but I cloned repo yesterday. 
I will try rebuild again because doesn't work now.

> $ darling shell
> Setting up a new Darling prefix at /home/xxxxx/.darling
> Cannot mount procfs: No such file or directory
> Cannot open mnt namespace file: No such file or directory

We does not use SeLinux ....

**Comment by blackPantherOS** [comment]
Two things...  I tested your RPM package, working fine on blackPanther OS as well, but only with my dkms module, because your dkms install progress build failed with following error:

```
make: Entering directory '/var/lib/dkms/darling-mach/0.1/build/lkm'
Running kernel version is 5.1.15-desktop-1bP
rm -f darling-mach.mod.o
make -C /lib/modules/5.1.15-desktop-1bP/build M=/var/lib/dkms/darling-mach/0.1/build/lkm modules
make[1]: Entering directory '/usr/src/kernel-5.1.15-desktop-1bP'
Running kernel version is 5.1.15-desktop-1bP
Invoked by kernel build system, building for 5.1.15-desktop-1bP
  CC [M]  /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/ipc/ipc_entry.o
  CC [M]  /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/ipc/ipc_hash.o
In file included from /var/lib/dkms/darling-mach/0.1/build/lkm/bsd/kern/ast.h:37,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/kern/processor.h:75,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/i386/cpu_data.h:41,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/machine/cpu_data.h:32,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/kern/cpu_data.h:42,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/kern/task.h:110,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/ipc/ipc_space.h:88,
                 from /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/ipc/ipc_hash.c:76:
/var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/kern/thread.h:434:2: error: unknown type name ‘task_watch_t’
  task_watch_t * taskwatch;  /* task watch */
  ^~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:276: /var/lib/dkms/darling-mach/0.1/build/lkm/osfmk/ipc/ipc_hash.o] Error 1
make[1]: *** [Makefile:1565: _module_/var/lib/dkms/darling-mach/0.1/build/lkm] Error 2
make[1]: Leaving directory '/usr/src/kernel-5.1.15-desktop-1bP'
make: *** [Makefile:315: default] Error 2
make: Leaving directory '/var/lib/dkms/darling-mach/0.1/build/lkm'

```
I can build successful with attached pacth:
[darling-kern.patch.gz](https://github.com/darlinghq/darling/files/4093599/darling-kern.patch.gz)

**Comment by ahyattdev** [comment]
@blackPantherOS is this the January 20, 2020 Alpha?
